src/server/req_delete.c

```diff
@@ -343,88 +381,54 @@ req_deletejob(struct batch_request *preq)
 		qdel_mail = 0;
 	else
 		qdel_mail = 1;
-	parent = chk_job_request(jid, preq, &jt);
-	if (parent == NULL)
+
+	if (FALSE == chk_job_request(jid, preq, &jt, &js, &pjob, &parent, &offset))
 		return; /* note, req_reject already called */
 
 	if (delhist) {
-		rc = check_deletehistoryjob(preq);
+		rc = check_deletehistoryjob(preq, pjob, parent, jt);
 		if (rc == TRUE)
 			return;
 	}
 
 
-	if (jt == IS_ARRAY_NO) {
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {
 
-		/* just a regular job, pass it on down the line and be done
+		/* just a regular job or a subjob, pass it on down the line and be done
 		 * If the request if to purge the history of the job then set ji_deletehistory to 1
 		 */
-		if (delhist)
-			parent->ji_deletehistory = 1;
-		req_deletejob2(preq, parent);
-		return;
+		if (pjob != NULL) {
+			if (delhist)
+				pjob->ji_deletehistory = 1;
+			req_deletejob2(preq, pjob);
 
-	} else if (jt == IS_ARRAY_Single) {
+			if (jt == IS_ARRAY_Single) {
 
-		/* single subjob, if running do full delete, */
-		/* if not then just set it expired		 */
+				/* single subjob, update tracking table header count */
 
-		offset = subjob_index_to_offset(parent, get_index_from_jid(jid));
-		if (offset == -1) {
-			req_reject(PBSE_UNKJOBID, 0, preq);
-			return;
-		}
-		i = get_subjob_state(parent, offset);
-		if (i == -1) {
-			req_reject(PBSE_IVALREQ, 0, preq);
-			return;
-		}
-		/*
-		 * If the request if to also purge the history of the sub job then log a message since
-		 * server does not store the history of subjobs
-		 */
-		if (delhist)
-			log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB,
-				LOG_INFO, jid,
-				msg_nosave_histarrayjob);
-
-		if (i == JOB_STATE_RUNNING) {
-			pjob = find_job(jid); /* get ptr to the subjob */
-			if (pjob) {
-				req_deletejob2(preq, pjob);
 				if (parent->ji_ajtrk)
 					if (pjob->ji_terminated)
 						parent->ji_ajtrk->tkm_dsubjsct++;
+
```

Comment:
If it is a subjob and somebody tries to delete its history then we used to get the following message in logs.
       "History of individual array subjobs not saved"
Now with this code change we no longer get this message. If this is desired then it is better to mention in EDD.

Reply:
Its reverted in rc 20 https://github.com/PBSPro/pbspro/pull/590/commits/b01fcac3e94363966f635e6fad726a5342288a79